### ISSUE TITLE
docs: Fetching favicons on every click in Chrome

### ISSUE THREAD

**Issue by onlyflix** [angular/angular#26544]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report  
[ ] Performance issue
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
[ ] Other... Please describe:
</code></pre>

## Current behavior

When I click anywhere on angular.io, the website always fetches all favicons (total of 19.3 KB).

## Expected behavior

The site gets them only once.

## Minimal reproduction of the problem with instructions

Go to https://angular.io, open the Network tab in the DevTools and click on the page. Only happens on Google Chrome.
## What is the motivation / use case for changing the behavior?

The website downloads way more content than it needs to.

## Environment

<pre><code>
Angular version: -

Browser:
- [x] Chrome (desktop) version 70.0.3538.67
- [ ] Chrome (Android) version XX
- [ ] Chrome (iOS) version XX
- [ ] Firefox version XX
- [ ] Safari (desktop) version XX
- [ ] Safari (iOS) version XX
- [ ] IE version XX
- [ ] Edge version XX
 
For Tooling issues:
- Node version: XX  
- Platform:  

Others:

</code></pre>

**Comment by skreborn** [comment]
For what it's worth, this doesn't happen if the user specifically clicks inside `aio-search-box`, presumably because it stops propagation. If you set up a click listener on every child of ˙aio-shell` that does the same, the issue goes away, leading to the conclusion that `aio-shell` should be the culprit. How or why, I have no idea just yet.

**Comment by skreborn** [comment]
A click on `aio-shell` closes the search box and calls `setSearch`, which in turn uses the native history API to `replaceState`, which triggers the behavior. Perhaps something is subscribed to the location change, that triggers this behavior.

**Comment by gkalpak** [comment]
Thx for reporting, @onlyflix 👍  Nice sleuthing, @skreborn :100:

It is this method (which is indeed called when we click anywhere in `aio-shell` outside of search results or search box.

https://github.com/angular/angular/blob/331989cea357abf9a20088c41a86a077435d17f7/aio/src/app/app.component.ts#L390-L393

We could easily change it like this:

```diff
-this.locationService.setSearch('', { ...this.locationService.search(), search: undefined });
+const oldSearch = this.locationService.search();
+if (oldSearch.search !== undefined) {
+  this.locationService.setSearch('', { ...oldSearch, search: undefined });
+}
```

@onlyflix or @skreborn: Feel free to submit a PR :wink:

**Comment by onlyflix** [comment]
Okay I found the reason for this issue. Like @skreborn already said, it is because of the native history API with replaceState. Chromium browser actually have a bug where the `window.history.replaceState()` call loads all favicons again. Refer to [this StackOverflow post](https://stackoverflow.com/questions/36103904/history-pushstate-in-chrome-make-favicon-request) and [this chromium issue](https://bugs.chromium.org/p/chromium/issues/detail?id=131567) for further information.

The StackOverflow post suggest to change the favicons to base64, which I tested. That makes the problem disappear.

@gkalpak With your solution, the favicons still get downloaded whenever I click on the Angular logo in the top left corner.

**Comment by skreborn** [comment]
@onlyflix While it's true that @gkalpak's solution still leaves room for this Chromium bug (that's supposedly fixed, by the way), it stilll produces ***much less *** erroneous requests. In any case, Chrome loads these favicons from disk cache according to the Network log, so I don't think there's much harm in it happening every once in a while.

**Comment by onlyflix** [comment]
I have to agree to that. The solution fixes most of the downloads, the rest is an issue on Chromiums site. 😉

But the bug actually isn't fixed, I did some more digging and found [this Chromium bug](https://bugs.chromium.org/p/chromium/issues/detail?id=895175), which is only 4 days old.

**Comment by skreborn** [comment]
@onlyflix Good find! Would you care to open a PR?

**Comment by onlyflix** [comment]
Just a quick follow up on the Chromium Bug:
This issue was fixed in Chrome Version 72.0.3623.0 with [this commit](https://chromium.googlesource.com/chromium/src/+/e9abde5a831c773c8e0ab85605254a591246925f). 😄
